packages/zoe/src/contracts/publicAuction.js

```diff
@@ -19,131 +21,106 @@ import '../../exported';
  *
- * publicAPI also has methods to find out what's being auctioned
- * (getAuctionedAssetsAmounts()), or the minimum bid (getMinimumBid()).
- *
- * @param {ContractFacet} zcf
+ * @type {ContractStartFn}
  */
-const makeContract = zcf => {
-  const { rejectOffer, satisfies, assertKeywords, checkHook } = makeZoeHelpers(
-    zcf,
-  );
-
-  let {
-    terms: { numBidsAllowed },
-  } = zcf.getInstanceRecord();
-  numBidsAllowed = Nat(numBidsAllowed !== undefined ? numBidsAllowed : 3);
-
-  let sellerOfferHandle;
+const start = zcf => {
+  const {
+    maths: { Asset: assetMath, Ask: bidMath },
+  } = zcf.getTerms();
+  let { numBidsAllowed = 3 } = zcf.getTerms();
+  numBidsAllowed = Nat(numBidsAllowed);
+
+  let sellSeat;
   let minimumBid;
   let auctionedAssets;
-  const allBidHandles = [];
+  const bidSeats = [];
 
   // seller will use 'Asset' and 'Ask'. buyer will use 'Asset' and 'Bid'
-  assertKeywords(harden(['Asset', 'Ask']));
+  assertIssuerKeywords(zcf, harden(['Asset', 'Ask']));
 
-  const bidderOfferHook = offerHandle => {
+  /** @type {OfferHandler} */
+  const bid = bidSeat => {
     // Check that the item is still up for auction
-    if (!zcf.isOfferActive(sellerOfferHandle)) {
+    if (sellSeat.hasExited()) {
       const rejectMsg = `The item up for auction is not available or the auction has completed`;
-      throw rejectOffer(offerHandle, rejectMsg);
+      throw bidSeat.kickOut(rejectMsg);
     }
-    if (allBidHandles.length >= numBidsAllowed) {
-      throw rejectOffer(offerHandle, `No further bids allowed.`);
+    if (bidSeats.length >= numBidsAllowed) {
+      throw bidSeat.kickOut(`No further bids allowed.`);
     }
-    const sellerSatisfied = satisfies(sellerOfferHandle, {
-      Ask: zcf.getCurrentAllocation(offerHandle).Bid,
-      Asset: zcf.getAmountMath(auctionedAssets.brand).getEmpty(),
+    const sellerSatisfied = satisfies(zcf, sellSeat, {
+      Ask: bidSeat.getAmountAllocated('Bid', minimumBid.brand),
+      Asset: assetMath.getEmpty(),
     });
-    const bidderSatisfied = satisfies(offerHandle, {
-      Asset: zcf.getCurrentAllocation(sellerOfferHandle).Asset,
-      Bid: zcf.getAmountMath(minimumBid.brand).getEmpty(),
+    const bidderSatisfied = satisfies(zcf, bidSeat, {
+      Asset: sellSeat.getAmountAllocated('Asset', auctionedAssets.brand),
+      Bid: bidMath.getEmpty(),
     });
     if (!(sellerSatisfied && bidderSatisfied)) {
       const rejectMsg = `Bid was under minimum bid or for the wrong assets`;
-      throw rejectOffer(offerHandle, rejectMsg);
+      throw bidSeat.kickOut(rejectMsg);
     }
 
     // Save valid bid and try to close.
-    allBidHandles.push(offerHandle);
-    if (allBidHandles.length >= numBidsAllowed) {
-      closeAuction(zcf, {
-        auctionLogicFn: secondPriceLogic,
-        sellerOfferHandle,
-        allBidHandles,
-      });
+    bidSeats.push(bidSeat);
+    if (bidSeats.length >= numBidsAllowed) {
+      closeAuction(zcf, secondPriceLogic, sellSeat, bidSeats);
```

Comment:
needs a comment since this is a big step of the contract.

Reply:
Also, I don' tthink all the exits and such should be in a helper. It's a pretty important thing andshould be real obvious